### PR TITLE
NS2.0 - Add SHA256/384/512CryptoServiceProvider

### PR DESCRIPTION
**Pull request by AtsushiKan**
Recent NetStandard 2.0 stragglers...

### CODE REVIEW

**Comment by joshfree** [comment]
Thanks @AtsushiKan.  LGTM.  @bartonjs PTAL.

File: src/System.Security.Cryptography.Csp/src/System/Security/Cryptography/SHA256CryptoServiceProvider.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        }
+
+        // The Hash and HashSize properties are not overridden since the correct values are returned from base.
+
+        protected sealed override void Dispose(bool disposing)
+        {
+            if (disposing)
+            {
+                _incrementalHash.Dispose();
+                base.Dispose(disposing);
```
Comment (stephentoub) [inline src/System.Security.Cryptography.Csp/src/System/Security/Cryptography/SHA256CryptoServiceProvider.cs]: Shouldn't base.Dispose(disposing) be called even if !disposing?

File: src/System.Security.Cryptography.Csp/src/System/Security/Cryptography/SHA256CryptoServiceProvider.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+// See the LICENSE file in the project root for more information.
+
+using System.ComponentModel;
+
+namespace System.Security.Cryptography
+{
+    //
+    // If you change this file, make the corresponding changes to all of the SHA*CryptoServiceProvider.cs files.
+    //
+    [EditorBrowsable(EditorBrowsableState.Never)]
```
Comment (stephentoub) [inline src/System.Security.Cryptography.Csp/src/System/Security/Cryptography/SHA256CryptoServiceProvider.cs]: Why are these marked as EditorBrowsableState.Never?

Comment (AtsushiKan) [inline src/System.Security.Cryptography.Csp/src/System/Security/Cryptography/SHA256CryptoServiceProvider.cs]: I think the idea is to discredit CSP, not just insecure algorithms (Aes hasn't been cracked afaik...)

**Comment by weshaggard** [comment]
cc @steveharter

**Pull request merged**